packages/browser/src/skylabClient.ts

```diff
@@ -106,43 +130,122 @@ export class SkylabClient implements Client {
+    } catch (e) {
+      console.error(e);
+      if (retry) {
+        this.startRetries();
       }
-      const endpoint = `${this.config.serverUrl}/sdk/vardata/${encodedContext}${queryString}`;
-      const headers = {
-        Authorization: `Api-Key ${this.apiKey}`,
+      return Promise.reject(e);
+    }
+  }
+
+  protected async doFetch(
+    user: SkylabUser,
+    timeoutMillis: number,
+  ): Promise<Response> {
+    const userContext = this.addContext(user);
+    const encodedContext = urlSafeBase64Encode(JSON.stringify(userContext));
+    let queryString = '';
+    let debugAssignmentRequestsParam;
+    if (this.debugAssignmentRequests) {
+      debugAssignmentRequestsParam = `d=${randomString(8)}`;
+    }
+    if (debugAssignmentRequestsParam) {
+      queryString = '?' + debugAssignmentRequestsParam;
+    }
+    const endpoint = `${this.config.serverUrl}/sdk/vardata/${encodedContext}${queryString}`;
+    const headers = {
+      Authorization: `Api-Key ${this.apiKey}`,
+    };
+    const response = await this.httpClient.requestWithTimeout(
+      timeoutMillis,
+      endpoint,
+      'GET',
+      headers,
+    );
+    if (this.debug) {
+      console.debug('[Skylab] Received fetch response:', response);
+    }
+    return response;
+  }
+
+  protected async parseResponse(response: Response): Promise<Variants> {
+    const json = await response.json();
+    const variants: Variants = {};
+    for (const flag of Object.keys(json)) {
+      variants[flag] = {
+        value: json[flag].key,
+        payload: json[flag].payload,
       };
-      const response = await this.httpClient.request(endpoint, 'GET', headers);
-      const json = await response.json();
-      this.storage.clear();
-      for (const flag of Object.keys(json)) {
-        this.storage.put(flag, {
-          value: json[flag].key,
-          payload: json[flag].payload,
-        });
-      }
-      this.storage.save();
+    }
+    if (this.debug) {
+      console.debug('[Skylab] Received variants:', variants);
+    }
+    return variants;
+  }
+
+  protected storeVariants(variants: Variants): void {
+    this.storage.clear();
+    for (const key in variants) {
+      this.storage.put(key, variants[key]);
+    }
+    this.storage.save();
+    if (this.debug) {
+      console.debug('[Skylab] Stored flags:', variants);
+    }
+  }
 
+  protected startRetries(): void {
+    if (this.debug) {
+      console.debug('[Skylab] Retry fetch all');
+    }
+    // Non-zero retry means we already have a retry interval in progress.
+    if (this.retry != 0) {
       if (this.debug) {
-        console.debug('[Skylab] Received and stored flags:', json);
+        console.debug('[Skylab] Retry fetch interval already in progress');
       }
-    } catch (e) {
-      console.error(e);
+      return;
     }
-    return this;
+    // Run fetchAll for the current user at an interval defined by the
+    // fetchRetryIntervalMillis config.
+    this.retry = window.setInterval(
```

Comment:
`globalThis.setInterval` returns a NodeJS `Timeout` type which I figured I should avoid. Do you know how to get around this? Kelvin mentioned I should use window to avoid nodejs specifics.